### PR TITLE
Feature: Poll Metrics Server Availability

### PR DESCRIPTION
**Pull request by UndeadRat22**

### CODE REVIEW

File: internal/services/controller/controller.go [hunk 1]
```diff
@@ -131,9 +135,10 @@ func New(
 		informers:       handledInformers,
 		agentVersion:    agentVersion,
 		healthzProvider: healthzProvider,
+		metricsClient:   metricsClient,
+		discovery:       discovery,
+		f:               f,
```
Comment (viktorasm) [inline internal/services/controller/controller.go:140]: nitpick: nicer name?

File: internal/services/controller/controller.go [hunk 2]
```diff
@@ -236,6 +241,29 @@ func (c *Controller) Run(ctx context.Context) error {
 		c.queue.ShutDown()
 	}()
 
+	podMetricsType := reflect.TypeOf(&v1beta1.PodMetrics{})
+	if c.informers[podMetricsType] == nil {
```
Comment (viktorasm) [inline internal/services/controller/controller.go:245]: will this ever be non-nil? `Run` only executes once right after "new"

File: internal/services/controller/controller.go [hunk 3]
```diff
@@ -236,6 +241,29 @@ func (c *Controller) Run(ctx context.Context) error {
...
+			if err := wait.PollInfiniteWithContext(ctx, time.Minute*2, func(ctx context.Context) (done bool, err error) {
+				if !isMetricsServerAPIAvailable(c.log, c.discovery) {
+					return false, nil
+				}
+
+				c.log.Infof("Cluster supports pod metrics, will start collecting them.")
+				metricsInformer := c.f.InformerFor(&v1beta1.PodMetrics{}, func(_ kubernetes.Interface, _ time.Duration) cache.SharedIndexInformer {
+					return custominformers.NewPodMetricsInformer(c.log, c.metricsClient)
+				})
+				informer := custominformers.NewHandledInformer(c.log, c.queue, metricsInformer, podMetricsType)
```
Comment (viktorasm) [inline internal/services/controller/controller.go:256]: duplicates logic in the "new" - maybe just remove from constructor, and leave only this delayed initialization?

File: internal/services/controller/controller.go (hunk 1 above)
Comment (UndeadRat22) [reply internal/services/controller/controller.go]: updated

File: internal/services/controller/controller.go (hunk 3 above)
Comment (UndeadRat22) [reply internal/services/controller/controller.go]: Removed the logic completely, will retest

**Review by viktorasm — APPROVED** [review]

**Pull request merged**